pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -535,6 +592,7 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 		// to return false, indicating that it has no more work to
 		// do.
 		info.host.ClearError()
+		info.host.RecordPoweredOn()
```

Comment:
I think you copied this form my earlier PR, but I just noticed two problems with it:

* This will get called multiple times, and update the timestamp each time when we only want it to update the timestamp the first time.
* Really we should write this (and verify that it is written) *before* starting to provision, power on, &c.

This implies that we also need to clear it when we have completed deprovisioning or power off.

Reply:
But if we clear `LastPoweredOn` after power off, this will break the existing code which relies on `lastPoweredOn.Before(pendingRebootSince)`  to detect reboot process.

Reply:
If the host is powered off we don't need to detect whether it needs to be rebooted, do we?

Reply:
We don't need to detect whether it needs to be rebooted but we used it to report that the power on was part of the reboot process. I guess we can give up on this. 
So if we put it before power on/provisioning - don't we take a risk that `LastPowerdOn` will be updated and the power on/provisioning action will fail for a while? This might be confusing

Reply:
Clearing `LastPoweredOn` would make things work but feels wrong IMHO.
Maybe its because the `Last` part in `LastPoweredOn` losses its relevance.

What about adding `LastPoweredOff` and only  updating `LastPoweredOn` if it's less than `LastPoweredOff` ?

Reply:
> Clearing LastPoweredOn would make things work but feels wrong IMHO.

Would renaming it to `PoweredOnTime` or `PoweredOnAt` make it feel less wrong? That would be simpler than adding another field.

> don't we take a risk that LastPowerdOn will be updated and the power on/provisioning action will fail for a while?

This is just the nature of the reconcile loop; you have to verify that you've successfully stored the state saying you're about to do something before you can start to do it. So not a big concern IMHO.

Reply:
I could like with changing it to ``PoweredOnAt``

Reply:
Great, just to be clear on this - `isInRebootProcess` will be true, when the host is powered off, even if we're not in a reboot process.
In other words, if we clear `PoweredOnAt` on power off, we don't have any way to differentiate between powered off machine that was powered off due to reboot or something else.

Reply:
The host has an online flag in the spec field to tell us if something is trying to force the power to be off. We aren't using that as part of the reboot, right?

Reply:
Can you please clarify the question? 
The reboot code never **modifies** the `Online` flag, but it can obviously power off the host as part of the reboot, even though that `Online` is `true`.

Reply:
Right, my point is that we can tell the difference between the states "off because we are rebooting" and "off because the user turned it off" by looking at spec.online.

Reply:
Correct. But we can't tell "on because we just ended a reboot" we just can say "on because user wants it on". Not a big deal maybe.

Reply:
If we can't tell that the host is on because just finished rebooting it, how do we know when to stop trying to reboot it?

Reply:
Since `PoweredOnAt` has been removed, this is outdated.